cmd/mongo/oplog_replay.go

```diff
@@ -0,0 +1,69 @@
+package mongo
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/databases/mongo"
+	"github.com/wal-g/wal-g/internal/databases/mongo/oplog"
+
+	"github.com/spf13/cobra"
+	"github.com/wal-g/tracelog"
+)
+
+func SetupSignalHandler(ctx context.Context, cancel func()) func() {
```

Comment:
Is there any analogue to Closeables in Java or Disposable in .Net?
Returning cleanup function is not a widely used idiom.
Here you have state of closing func as local variable ch